include/links-model.php

```diff
@@ -147,64 +148,35 @@ public function get_hosts() {
 	 * Returns the home url in a given language.
 	 *
 	 * @since 1.3.1
+	 * @since 3.4 Accepts now a language slug.
 	 *
-	 * @param PLL_Language $lang The language object.
+	 * @param PLL_Language|string $language Language object or slug.
 	 * @return string
 	 */
-	public function home_url( $lang ) {
-		$url = trailingslashit( $this->home );
-		return $this->options['hide_default'] && $lang->slug == $this->options['default_lang'] ? $url : $this->add_language_to_link( $url, $lang );
-	}
+	public function home_url( $language ) {
+		if ( $language instanceof PLL_Language ) {
+			$language = $language->slug;
+		}
 
-	/**
-	 * Sets the home urls in PLL_Language.
-	 *
-	 * @since 1.8
-	 *
-	 * @param PLL_Language $language The language object.
-	 * @return void
-	 */
-	protected function set_home_url( $language ) {
-		$search_url = $this->home_url( $language );
-		$home_url = empty( $language->page_on_front ) || $this->options['redirect_lang'] ? $search_url : $this->front_page_url( $language );
-		$language->set_home_url( $search_url, $home_url );
-	}
+		$url = trailingslashit( $this->home );
 
-	/**
-	 * Sets the home urls before the languages are persistently cached.
-	 *
-	 * @since 1.8
-	 *
-	 * @param PLL_Language[] $languages Array of PLL_Language objects.
-	 * @return PLL_Language[] Array of PLL_Language objects with home url.
-	 */
-	public function pll_languages_list( $languages ) {
-		foreach ( $languages as $language ) {
-			$this->set_home_url( $language );
-		}
-		return $languages;
+		return $this->options['hide_default'] && $language === $this->options['default_lang'] ? $url : $this->add_language_to_link( $url, $language );
 	}
 
 	/**
-	 * Sets the home urls when not cached.
-	 * Sets the home urls scheme.
+	 * Adds home and search URLs to language data before the object is created.
 	 *
-	 * @since 1.8
+	 * @since 3.4
 	 *
-	 * @param PLL_Language[] $languages Array of PLL_Language objects.
-	 * @return PLL_Language[] Array of PLL_Language objects.
+	 * @param array $default  Array of language default data values containing home and search URLs.
+	 * @param array $language Language data.
+	 * @return array Language data with URLs addeddefault.
 	 */
-	public function pll_after_languages_cache( $languages ) {
-		foreach ( $languages as $language ) {
-			// Get the home urls when not cached.
-			if ( ( defined( 'PLL_CACHE_LANGUAGES' ) && ! PLL_CACHE_LANGUAGES ) || ( defined( 'PLL_CACHE_HOME_URL' ) && ! PLL_CACHE_HOME_URL ) ) {
-				$this->set_home_url( $language );
-			}
+	public function set_language_urls( $default, $language ) {
```

Comment:
I'd like to keep `home` in the method name. I suggest `set_home_urls()` or `set_language_home_urls()`.

Reply:
Done in 42de22a.